lib/validate.js

```diff
@@ -50,6 +50,7 @@ module.exports = function makeValidatorMiddleware (middleware, schema, opts) {
           p.required && !reqSchema.properties.query.required.includes(p.name) && reqSchema.properties.query.required.push(p.name)
           break
         case 'header':
+          // eslint-disable-next-line no-case-declarations
```

Comment:
Why are you ignoring `no-case-declarations` for the next line here? Looking at the [eslint docs for this lint](https://eslint.org/docs/latest/rules/no-case-declarations), it doesn't seem to pertain to anything here?

Unless I'm misunderstanding something here, I'd suggest to remove this line.

```suggestion
```

Reply:
Quoting the eslint docs:
> This rule disallows lexical declarations (let, const, function and class) in case/default clauses. The reason is that the lexical declaration is visible in the entire switch block but it only gets initialized when it is assigned, which will only happen if the case where it is defined is reached.
>
> To ensure that the lexical declaration only applies to the current case clause wrap your clauses in blocks.
There is a constant declaration on L54. To ensure that the declaration does not leak to out of the case block, I would personally wrapped the case block with braces, like this:

```
case 'header': {
  const name = p.name.toLowerCase()
  reqSchema.properties.headers.properties[name] = p.schema
  p.required && !reqSchema.properties.headers.required.includes(p.name) && reqSchema.properties.headers.required.push(name)
  break
}
```
However, the [author chose to ignore the rule](https://github.com/Unleash/express-openapi/commit/98add2cbd89818cd90d43aa0d6cc8d6733dca1fe#diff-cabded08ff9373812a499e53913ccaca1d50f6d981eefa5c45f8cb88809a9cf7R53).

Reply:
Feel free to open a pr if y'all want. As with any opinionated lint rules, they are there to protect against making mistakes without realizing it. I for sure think moving it out and using `let` is worse since it would separate it from the context in which it was used. Linting would also catch it leaking, so I don't think it matters, but would totally accept a PR adding the `{}` since that is more explicit about the intent anyway.

Reply:
No offense @wesleytodd, if I sounded bad :pray: and thank you  for chiming in. I appreciate your hard work.
And yes, I don't think it matters either; that's why I left it as is. Once this is merged, I'll do a pull request to both repos.
Cheers!

Reply:
Oh I didn't take offense. It just code, lots of ways to approach problems. I just figured an explanation of the thinking behind it might help. And I would love a PR with stuff y'all have found using the lib!

Reply:
Sounds great! And yeah, sorry, my mind must've been somewhere else when I was looking through this. I didn't register that the change was in a switch block at all. Yeah, given that, it's totally fine 😅